traceapp/dashboard.go

```diff
@@ -145,13 +145,43 @@ func (a *App) serveDashboardData(w http.ResponseWriter, r *http.Request) error {
 	// Important: If it is a nil slice it will be encoded to JSON as null, and the
 	// bootstrap-table library will not update the table with "no entries".
 	rows := make([]dashboardRow, 0)
+	var rowsErr error
+
+	// Uses the appropriate strategy to obtain the `rows` to be returned based on the Queryer's type,
+	// this because `InfluxDBStore` & `AggregateStore` handles aggregation logic differently:
+	// `Aggregatestore` uses traces to store aggregation information whereas `InfluxDBStore` performs
+	// sql-like queries to obtain traces which are used to obtain aggregation information.
+	switch a.Queryer.(type) {
+	case *appdash.InfluxDBStore:
+		rows, rowsErr = influxDBStoreRows(traces, tracesURL)
+	default:
+		rows, rowsErr = aggregateStoreRows(traces, start, end, tracesURL)
+	}
```

Comment:
I think we do not need to keep compatibility for `AggregateStore` anymore because `InfluxDBStore` will replace it entirely. Could we just remove `AggregateStore` entirely as part of this PR?